### PR TITLE
homework01 - Jisung Lee

### PR DESCRIPTION
**Pull request by jskurt**
first PR from homework/jisung-lee

### CODE REVIEW

**Comment by jskurt** [comment]
[add to setTodo routine in handleCancel is it right...?]
여기까지해도 기능은되는거같은데...틀렸다해서 뭐가 잘못됬나 궁금,,, 훔...

**Comment by kafelix496** [comment]
> [add to setTodo routine in handleCancel is it right...?]
> 
> 여기까지해도 기능은되는거같은데...틀렸다해서 뭐가 잘못됬나 궁금,,, 훔...

edit 을 하다가 뭔가 수정을 하고나서 cancel을 하고, 다시 edit 을 눌릴경우, 값이 리셋되어야함.

File: src/components/containers/Todo.js [hunk 1]
```diff
@@ -4,22 +4,66 @@ import TodoTop from '../presentations/TodoTop'
...
   }
   const deleteTodo = (id) => {
     setTodos((prev) => prev.filter((todo) => todo.id !== id))
   }
+  // This is only active test
+  // const test = () => {
+  //   console.log(todos)
+  // }
+  const editStateCheck = (id) => {
+    setTodos(todos.map((todo) => (todo.editState = false)))
```
Comment (kafelix496) [inline src/components/containers/Todo.js:19]: What is this line?

File: src/components/containers/Todo.js [hunk 2]
```diff
@@ -4,22 +4,66 @@ import TodoTop from '../presentations/TodoTop'
 
 function Todo() {
   const [todos, setTodos] = useState([])
+  const [edit, setEdit] = useState()
```
Comment (kafelix496) [inline src/components/containers/Todo.js:7]: This is not necessary.

File: src/components/containers/Todo.js [hunk 3]
```diff
@@ -4,22 +4,66 @@ import TodoTop from '../presentations/TodoTop'
...
   const addTodo = (todo) => {
     setTodos([...todos, todo])
   }
   const deleteTodo = (id) => {
     setTodos((prev) => prev.filter((todo) => todo.id !== id))
   }
+  // This is only active test
+  // const test = () => {
+  //   console.log(todos)
+  // }
```
Comment (kafelix496) [inline src/components/containers/Todo.js:17]: Do not leave a comment like this.

File: src/components/containers/Todo.js (hunk 1 above)
Comment (kafelix496) [reply src/components/containers/Todo.js:19]: Also, as I said before, you must not mutate the original value.

File: src/components/containers/Todo.js [hunk 4]
```diff
@@ -4,22 +4,66 @@ import TodoTop from '../presentations/TodoTop'
...
+  }
+
+  const cancelCheck = (id) => {
+    setTodos(
+      todos.map((todo) =>
+        todo.id === id ? { ...todo, editState: !todo.editState } : todo
+      )
+    )
+
+    setEdit(false)
```
Comment (kafelix496) [inline src/components/containers/Todo.js:37]: > const [edit, setEdit] = useState()

This should be removed with the code above.

File: src/components/containers/Todo.js [hunk 5]
```diff
@@ -4,22 +4,66 @@ import TodoTop from '../presentations/TodoTop'
...
+
+  const changeTodo = (id, textItem) => {
+    setTodos(
+      todos.map((todo) =>
+        todo.id === id
+          ? { ...todo, text: textItem, editState: !todo.editState }
+          : todo
+      )
+    )
+    setEdit(false)
```
Comment (kafelix496) [inline src/components/containers/Todo.js:48]: > const [edit, setEdit] = useState()

This should be removed with the code above.

File: src/components/presentations/TodoItem.js [hunk 6]
```diff
@@ -1,14 +1,70 @@
...
+
+  const handleDelete = () => {
+    deleteTodo(id)
+  }
+  const handleChange = (event) => {
+    setTodo(event.target.value)
+  }
+  // const onClick = () => {
+  //   test()
+  // }
```
Comment (kafelix496) [inline src/components/presentations/TodoItem.js:42]: Do not leave a comment like this. Remove it.

File: src/components/presentations/TodoItem.js [hunk 7]
```diff
@@ -1,14 +1,70 @@
...
+
+      {archive && <button onClick={handleDelete}>delete</button>}
+
+      {editState && (
+        <>
+          <button onClick={handleConfirm}>confirm</button>
+          <button onClick={handleCancel}>cancel</button>
+        </>
+      )}
+      {/* <button onClick={onClick}>test</button> */}
```
Comment (kafelix496) [inline src/components/presentations/TodoItem.js:67]: Remove the test code when you finish your work.

**Review by kafelix496 — COMMENTED** [review]
Think about how to check if one of the to-do items is editing without using another state. This would trigger unnecessary re-render.

File: src/components/containers/Todo.js (hunk 3 above)
Comment (jskurt) [reply src/components/containers/Todo.js:17]: ok
 i got it

File: src/components/presentations/TodoItem.js (hunk 7 above)
Comment (jskurt) [reply src/components/presentations/TodoItem.js:67]: yas

File: src/components/containers/Todo.js (hunk 1 above)
Comment (jskurt) [reply src/components/containers/Todo.js:15]: 원래값 공유 안하고 생각해보겠음
새로 복사해야하는건가

File: src/components/containers/Todo.js (hunk 2 above)
Comment (jskurt) [reply src/components/containers/Todo.js:7]: ok i deleted

File: src/components/containers/Todo.js (hunk 3 above)
Comment (jskurt) [reply src/components/containers/Todo.js]: i deleted

File: src/components/containers/Todo.js (hunk 4 above)
Comment (jskurt) [reply src/components/containers/Todo.js:33]: i deleted

File: src/components/containers/Todo.js (hunk 5 above)
Comment (jskurt) [reply src/components/containers/Todo.js:44]: i deleted also

File: src/components/presentations/TodoItem.js (hunk 6 above)
Comment (jskurt) [reply src/components/presentations/TodoItem.js]: i deleted also

File: src/components/containers/Todo.js (hunk 2 above)
Comment (kafelix496) [reply src/components/containers/Todo.js:7]: no you didn't delete it

**Review by kafelix496 — COMMENTED** [review]
Press the 'resolve' button if you resolved it

**Comment by jskurt** [comment]
> Press the 'resolve' button if you resolved it

i didnt, then where is resolve button?

**Comment by jskurt** [comment]
> Press the 'resolve' button if you resolved it

ahhhhh i find it

File: src/index.js [hunk 8]
```diff
@@ -6,9 +6,9 @@ import reportWebVitals from './reportWebVitals'
 
 const root = ReactDOM.createRoot(document.getElementById('root'))
 root.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>
+  // <React.StrictMode>
+  <App />
+  // </React.StrictMode>
```
Comment (kafelix496) [inline src/index.js:11]: Use strict mode.

**Comment by jskurt** [comment]
setTodos를 하나만 쓰고싶은데 혹시 불가한가요? 제 visualstudio code에서는 저 쉼표로 구분하니 되던데,, git에서는 에러뜨네요 흠

**Comment by kafelix496** [comment]
mission complete. well done.

**Pull request closed without merging**